### PR TITLE
update k8s reference

### PR DESCRIPTION
**Pull request by wasaga**

### CODE REVIEW

**Review by desimone — COMMENTED** [review]
Just to confirm you are using prettier to format all these files? I'm trying to limit the auto-format based conflicts. otherwise, lgtm pending @ZPain8464

**Comment by wasaga** [comment]
These files are auto generated

**Comment by ZPain8464** [comment]
Reviewing now, sorry for the delay here.

File: src/ingress.pomerium.io_pomerium.yaml [hunk 1]
```diff
@@ -14,262 +14,281 @@ spec:
...
+                  type: string
+                type: array
+              cookie:
+                description: Cookie defines Pomerium cookie options
+                properties:
+                  domain:
+                    description: Domain see https://docs.pomerium.com/docs/reference/cookie-domain
+                    type: string
+                  expire:
+                    description: Expire see https://docs.pomerium.com/docs/reference/cookie-expire
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:65]: This link 404s. Is there a reason we're the subdomain path "docs.pomerium.com/docs/" and not just "pomerium.com/docs/"?

File: src/ingress.pomerium.io_pomerium.yaml [hunk 2]
```diff
@@ -14,262 +14,281 @@ spec:
...
+                    type: string
+                  httpOnly:
+                    description: HTTPOnly see https://docs.pomerium.com/docs/reference/cookie-http-only
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:68]: 404

File: src/ingress.pomerium.io_pomerium.yaml [hunk 3]
```diff
@@ -14,262 +14,281 @@ spec:
...
+                    type: boolean
+                  name:
+                    description: Name see https://docs.pomerium.com/docs/reference/cookie-name
+                    type: string
+                  secure:
+                    description: Secure see https://docs.pomerium.com/docs/reference/cookie-secure
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:74]: This link 404s

File: src/ingress.pomerium.io_pomerium.yaml [hunk 4]
```diff
@@ -14,262 +14,281 @@ spec:
...
+                    type: boolean
+                type: object
+              identityProvider:
+                description: IdentityProvider see https://www.pomerium.com/docs/identity-providers/
+                properties:
+                  provider:
+                    description: Provider one of accepted providers - see https://www.pomerium.com/reference/#identity-provider-name.
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:81]: Small detail, but I'd remove the "." at the end of the link

File: src/ingress.pomerium.io_pomerium.yaml [hunk 5]
```diff
@@ -14,262 +14,281 @@ spec:
...
-                      format: uri
-                      pattern: ^https://
+                    description: RequestParams see https://www.pomerium.com/reference/#identity-provider-request-params
+                    type: object
+                  requestParamsSecret:
+                    description: RequestParamsSecret is a reference to a secret for
+                      additional parameters you'd prefer not to provide in plaintext.
+                    type: string
+                  scopes:
+                    description: Scopes see https://www.pomerium.com/reference/#identity-provider-scopes.
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:119]: the /reference/ links in lines 112 and 119 both redirect to the main reference page: https://www.pomerium.com/docs/reference

Correct RequestParams link: https://www.pomerium.com/docs/reference/identity-provider-request-params
Correct IdP Scopes link: https://www.pomerium.com/docs/reference/identity-provider-scopes

File: src/ingress.pomerium.io_pomerium.yaml [hunk 6]
```diff
@@ -14,262 +14,281 @@ spec:
...
-                secrets:
-                  description:
-                    Secrets references a Secret that must have the following
-                    keys - shared_secret - cookie_secret - signing_key
-                  minLength: 1
+                    type: array
+                  secret:
+                    description: Secret containing IdP provider specific parameters
+                      and must contain at least client_id and client_secret values,
+                      an optional `service_account` field, mapped to https://www.pomerium.com/reference/#identity-provider-service-account
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:126]: I can't find an identity provider service account page?

File: src/ingress.pomerium.io_pomerium.yaml [hunk 7]
```diff
@@ -14,262 +14,281 @@ spec:
...
+                    minLength: 1
+                    type: string
+                  serviceAccountFromSecret:
+                    description: ServiceAccountFromSecret is no longer supported,
+                      see https://docs.pomerium.com/docs/overview/upgrading#idp-directory-sync
+                    type: string
+                  url:
+                    description: URL is identity provider url, see https://www.pomerium.com/reference/#identity-provider-url.
```
Comment (ZPain8464) [inline src/ingress.pomerium.io_pomerium.yaml:134]: Correct link: https://docs.pomerium.com/docs/reference/identity-provider-url

File: src/k8sSettings.json [hunk 8]
```diff
@@ -62,11 +64,37 @@
...
                   },
+                  "cookie": {
+                    "description": "Cookie defines Pomerium cookie options",
+                    "properties": {
+                      "domain": {
+                        "description": "Domain see https://docs.pomerium.com/docs/reference/cookie-domain",
+                        "type": "string"
+                      },
+                      "expire": {
+                        "description": "Expire see https://docs.pomerium.com/docs/reference/cookie-expire",
```
Comment (ZPain8464) [inline src/k8sSettings.json:75]: cookie-expire link 404s

File: src/k8sSettings.json [hunk 9]
```diff
@@ -62,11 +64,37 @@
...
+                        "type": "string"
+                      },
+                      "httpOnly": {
+                        "description": "HTTPOnly see https://docs.pomerium.com/docs/reference/cookie-http-only",
```
Comment (ZPain8464) [inline src/k8sSettings.json:79]: Link 404s

File: src/k8sSettings.json [hunk 10]
```diff
@@ -62,11 +64,37 @@
...
+                        "type": "boolean"
+                      },
+                      "name": {
+                        "description": "Name see https://docs.pomerium.com/docs/reference/cookie-name",
+                        "type": "string"
+                      },
+                      "secure": {
+                        "description": "Secure see https://docs.pomerium.com/docs/reference/cookie-secure",
```
Comment (ZPain8464) [inline src/k8sSettings.json:87]: Link 404s

File: src/k8sSettings.json [hunk 11]
```diff
@@ -62,11 +64,37 @@
...
+                      }
+                    },
+                    "type": "object"
+                  },
                   "identityProvider": {
                     "description": "IdentityProvider see https://www.pomerium.com/docs/identity-providers/",
                     "properties": {
                       "provider": {
-                        "description": "Provider one of accepted providers https://www.pomerium.com/reference/#identity-provider-name",
+                        "description": "Provider one of accepted providers - see https://www.pomerium.com/reference/#identity-provider-name.",
```
Comment (ZPain8464) [inline src/k8sSettings.json:97]: This also redirects to the reference homepage. 

Correct link: https://www.pomerium.com/docs/reference/identity-provider-name

File: src/k8sSettings.json [hunk 12]
```diff
@@ -102,29 +135,43 @@
                         "type": "object"
                       },
                       "requestParamsSecret": {
-                        "description": "RequestParamsSecret is a reference to a secret for additional parameters you'd prefer not to provide in plaintext",
+                        "description": "RequestParamsSecret is a reference to a secret for additional parameters you'd prefer not to provide in plaintext.",
                         "type": "string"
                       },
                       "scopes": {
-                        "description": "Scopes see https://www.pomerium.com/reference/#identity-provider-scopes",
+                        "description": "Scopes see https://www.pomerium.com/reference/#identity-provider-scopes.",
```
Comment (ZPain8464) [inline src/k8sSettings.json:142]: Correct link: https://www.pomerium.com/docs/reference/identity-provider-scopes

File: src/k8sSettings.json [hunk 13]
```diff
@@ -102,29 +135,43 @@
...
                         "items": {
                           "type": "string"
                         },
                         "type": "array"
                       },
                       "secret": {
-                        "description": "Secret refers to a k8s secret containing IdP provider specific parameters and must contain at least `client_id` and `client_secret` map values",
+                        "description": "Secret containing IdP provider specific parameters and must contain at least client_id and client_secret values, an optional `service_account` field, mapped to https://www.pomerium.com/reference/#identity-provider-service-account",
```
Comment (ZPain8464) [inline src/k8sSettings.json:149]: As the link is, it redirects to reference homepage. I tried correcting it:  https://www.pomerium.com/reference/identity-provider-service-account

But, it's 404ing

File: src/k8sSettings.json [hunk 14]
```diff
@@ -102,29 +135,43 @@
...
                         "minLength": 1,
                         "type": "string"
                       },
+                      "serviceAccountFromSecret": {
+                        "description": "ServiceAccountFromSecret is no longer supported, see https://docs.pomerium.com/docs/overview/upgrading#idp-directory-sync",
+                        "type": "string"
+                      },
                       "url": {
-                        "description": "URL is identity provider url, see https://www.pomerium.com/reference/#identity-provider-url",
+                        "description": "URL is identity provider url, see https://www.pomerium.com/reference/#identity-provider-url.",
```
Comment (ZPain8464) [inline src/k8sSettings.json:158]: Correct link: https://www.pomerium.com/docs/reference/identity-provider-url

**Review by ZPain8464 — COMMENTED** [review]
@desimone and @wasaga I went through all the links and found many that 404 or redirect to the reference homepage. I commented on all of them and tried including the right link, but if this behavior is expected for some reason or I'm not aware, just please let me know for next time. 

I noticed we discussed this docs.pomerium and pomerium.com/docs in the #docs Slack channel, but not sure what the consensus is. Best practice for SEO is to use a /subfolder, but open to whatever the team needs here.

**Comment by wasaga** [comment]
superseded by https://github.com/pomerium/documentation/pull/160

**Pull request closed without merging**